Rating Page

butalas opened the pull request:
### Description
This branch designs the rating page.

### Context
It mimics the designs from Figma as one of the screens for the app. I wasn't sure whether to do the tool bar so I didn't implement that on the screen.

### Issue
#27 

### Screenshot
![img_6058](https://user-images.githubusercontent.com/30233987/52888560-21727180-314a-11e9-9e7e-705f36b2a2e1.PNG)

ijemmao commented on App.js:
```diff
@@ -1,11 +1,44 @@
 import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import { StyleSheet, Text, View, Dimensions, Image} from 'react-native';
```
Revert your App.js file back to what it looks like on the `master` branch

ijemmao commented on screens/login.js:
```diff
@@ -1,16 +1,18 @@
 import React from 'react';
```
It looks like you checked out this branch from your `login` branch - please make sure that you checkout only from the `master` branch. If you don't, you include code that isn't technically approved to be merged into another.

Remove at files that are connected to login - that includes other js or assets files.

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,160 @@
+  };
+  async componentDidMount() {
+    await Font.loadAsync({
+      'Montserrat-Regular': require('/Users/nona/shopu/assets/fonts/Montserrat-Regular.otf'),
```
Again, change the paths of your fonts

butalas replied:
When I change the paths and reload the app, it says it's unable to find the fonts. Do I have to put in exactly `./../assets/fonts/Montserrat-Regular.otf`?

ijemmao replied:
No, that was just an example

When I was walking through bash commands, I mentioned that `.` is the current directory and `..` is the parent directory.

butalas replied:
I got it to work! Thanks!

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,160 @@
+            <Text style={styles.usingText}>using</Text>
+            <Text style={styles.shopuText}>shopU!</Text>
+            <View style={styles.ratingBox} />
+            <Image style={styles.ratingIcons}
```
This can't just be images - this needs to be individual stars

Instead of having just an `Image` component, you should use the `View` component, and set its background to the star image.

butalas replied:
So how do I split the stars from that image? I saved the rating_group.png as a component from Figma. Do I need to make 20 Views?

butalas commented:
I addressed your comments on the rating page. I used margins for the stars, but the remaining components on the screen are still positioned absolutely.

ijemmao commented:
@butalas This is looking really good - the only thing I want you to do is remove the `Font.loadAsync()` code from the rating page.

**Before you do that**: on your local `ratingPage` branch call `git pull origin ratingPage --force` this will sync up your local with the remote version

butalas commented:
Sorry I haven't been on campus all day so I just looked at the font thing. I removed the `Font.loadAsync()` code from the rating page like you asked.

ijemmao commented on screens/buyer_search.js:
```diff
@@ -0,0 +1,206 @@
+import React from 'react';
```
This file shouldn't be included in this PR, you can remove it

ijemmao commented on screens/login.js:
```diff
@@ -1,38 +1,24 @@
 import React from 'react';
```
Remove this file

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,224 @@
+            <Text style={styles.shopuText}>shopU!</Text>
+            <View style={styles.ratingBox} />
+            <Text style={styles.delivererText}>Deliverer:</Text>
+            <View style={{width: 300, flexDirection: 'row', marginTop: 800, marginLeft: 185}}>
```
You shouldn't have inline styles - make a new style below

Also use flexboxes - you shouldn't be using `800px` margins to position items

butalas replied:
Should I make it come down like 3/4 of the page (something along those lines)?

butalas replied:
Like a percentage instead of pixels

ijemmao replied:
I would first use flexboxes first and then use margins to tweak the small details

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,224 @@
+              {this.stars()}
+              </View>
+            <Text style={styles.timingText}>Timing:</Text>
+            <View style={{width: 300, flexDirection: 'row', marginTop: 65, marginLeft: 185}}>
```
This line is repeated three times - create a style

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,224 @@
+    return (
+      <View>
+      {
+        this.state.fontLoaded ? (
```
Remote this line - `App.js` already checks for the font

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,224 @@
+  state = {
+    fontLoaded: false,
+  };
+  componentWillMount = () => {
```
Remove the `componentWillMount` function

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,224 @@
+const window = Dimensions.get("window")
+
+export default class Rating extends React.Component {
+  state = {
```
Remove this `state` variable

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import { StyleSheet, Text, View, Dimensions, Image} from 'react-native';
+import { Font } from 'expo';
+import FontLoad from './components/FontLoad';
```
Remove this line

ijemmao commented on screens/rating.js:
```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import { StyleSheet, Text, View, Dimensions, Image} from 'react-native';
+import { Font } from 'expo';
```
Remove this line - `App.js` should be the only file with references to `Font`

butalas commented:
I just addressed all the comments you made regarding the rating page!
